### PR TITLE
[#10950] Add base files for audit logs backend

### PR DESCRIPTION
**Pull request by AdithyaNarayan**
Part of #10950 

Added base files for the audit logs
- `FeedbackResponseAccessType`, `FeedbackResponseLog`, `FeedbackResponseLogsData`, `StudentResponseAccessLog` in the `output` package for API Output
- New endpoint `webapi/sesion/accesslogs` and `GetSessionAccessLogsAction`

### CODE REVIEW

**Comment by t-cheepeng** [comment]
On hold until tracking of access and submission is possible

File: src/main/java/teammates/ui/output/FeedbackResponseLog.java
Comment (halfwhole) [inline src/main/java/teammates/ui/output/FeedbackResponseLog.java:8]: Perhaps `FeedbackSessionLog`? (I'm not really sure what response log means in this context!)

File: src/main/java/teammates/ui/output/StudentResponseAccessLog.java
Comment (halfwhole) [inline src/main/java/teammates/ui/output/StudentResponseAccessLog.java:19]: Looks good to me, though it's probably down to variable namings again!
Perhaps the `FeedbackResponseAccessType` could be something like `FeedbackSessionLogType` (feels more natural to me)? Or `accessTimestamp` as simply `timestamp` and `StudentResponseAccessLog` as `FeedbackSessionLogEntry` etc.?

File: src/main/java/teammates/common/util/Const.java
Comment (AdithyaNarayan) [reply src/main/java/teammates/common/util/Const.java]: I was thinking access can be either a visit or a submission. I thought that just calling it logs might be misleading since its only the logs that we are writing and not the request logs. I will change to logs as it makes more sense.

**Comment by AdithyaNarayan** [comment]
Changed some of the naming

**Comment by AdithyaNarayan** [comment]
Can this PR be removed from hold, now that the endpoint is up? Would help in generating types to call the create logs endpoint

File: .github/workflows/component.yml
Comment (halfwhole) [inline .github/workflows/component.yml:13]: How did these get here :o

File: src/main/java/teammates/ui/webapi/GetSessionAccessLogsAction.java
Comment (halfwhole) [inline src/main/java/teammates/ui/webapi/GetSessionAccessLogsAction.java:15]: Missed this earlier, but is it better for it to be `AuthType.LOGGED_IN`? Can't think of a situation where the instructor can get the logs without being logged in first

File: src/main/java/teammates/ui/output/FeedbackSessionLogType.java
Comment (halfwhole) [inline src/main/java/teammates/ui/output/FeedbackSessionLogType.java:16]: Currently from `Const.FeedbackSessionLogTypes` we already have `ACCESS` and `SUBMISSION`, so there seems to be duplication with what we have over here. If we import those constants over and use them here through an enum constructor, such as `SUBMISSION(Const.FeedbackSessionLogTypes.SUBMISSION)`, would it be better?

Comment (AdithyaNarayan) [reply src/main/java/teammates/ui/output/FeedbackSessionLogType.java]: I have created a new enum `LogTypes` in the #11005  that is currently being used for generating types for the frontend. That can be reused here once that PR has been merged

Comment (AdithyaNarayan) [reply src/main/java/teammates/ui/output/FeedbackSessionLogType.java]: Have removed this for now, thanks!

**Review by halfwhole — APPROVED** [review]
LGTM, not sure why the component tests failed in installing the dependencies

**Review by t-cheepeng — APPROVED** [review]

File: src/main/java/teammates/ui/output/FeedbackSessionLogEntry.java
Comment (madanalogy) [inline src/main/java/teammates/ui/output/FeedbackSessionLogEntry.java:26]: Any particular reason is this marked as a TODO? What preventing the use of `LogType`?

File: src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java
Comment (madanalogy) [inline src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java:34]: Shouldn't there be a TODO here pending further implementation development?

**Review by madanalogy — COMMENTED** [review]
Base files look alright, just a couple of questions below. Also, probably can reword the title of the PR to be more appropriate (e.g. "Add GET endpoint for audit logs backend")

**Pull request merged**
